### PR TITLE
ER 677 Moves & Renames

### PR DESCRIPTION
**Pull request by leewadhams**
Main changes are in the Infrastructure project.

* Reference data classes moved out into own folders for each type.
* Providers moved out of services and Services renamed Providers where applicable.
* Move ApprenticeshipProgrammes to be stored in Reference Data collection.

### CODE REVIEW

File: src/Shared/Recruit.Vacancies.Client/Application/Services/SlaService.cs [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 {
     public class SlaService : ISlaService
     {
         private const int SlaHours = 24;
 
-        private readonly IBankHolidayService _bankholidayService;
+        private readonly IBankHolidayProvider _bankholidayProvider;
 
-        public SlaService(IBankHolidayService bankholidayService)
+        public SlaService(IBankHolidayProvider bankholidayService)
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Application/Services/SlaService.cs]: IBankHolidayProvider bankholiday**Provider**

File: src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -83,6 +90,8 @@ private static void RegisterServiceDeps(IServiceCollection services, IConfigurat
             services.AddTransient<IGeocodeServiceFactory, GeocodeServiceFactory>();
 
             services.Configure<FaaConfiguration>(configuration.GetSection("FaaConfiguration"));
+            
+            services.AddApprentieshipsApi(configuration);
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs]: AddApprenti**c**eshipsApi

File: src/Shared/UnitTests/Vacancies.Client/Infrastructure/Providers/NationalMinimumWageProviderTests.cs [hunk 3]
```diff
@@ -11,18 +10,18 @@
 
 namespace Esfa.Recruit.Vacancies.Client.UnitTests.Vacancies.Client.Infrastructure.Services
 {
-    public class NationalMinimumWageServiceTests
+    public class NationalMinimumWageProviderTests
     {
-        private readonly NationalMinimumWageService _service;
-        private readonly Mock<ILogger<NationalMinimumWageService>> _mockLogger;
+        private readonly NationalMinimumWageProvider _service;
```
Comment (chrisjensenuk) [inline src/Shared/UnitTests/Vacancies.Client/Infrastructure/Providers/NationalMinimumWageProviderTests.cs]: NationalMinimumWageProvider _provider

File: src/Shared/Recruit.Vacancies.Client/Application/Services/SlaService.cs (hunk 1 above)
Comment (leewadhams) [reply src/Shared/Recruit.Vacancies.Client/Application/Services/SlaService.cs]: 👍

File: src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (leewadhams) [reply src/Shared/Recruit.Vacancies.Client/Ioc/ServiceCollectionExtensions.cs]: 👍

**Comment by leewadhams** [comment]
@chrisjensenuk The ApprenticeshipProgrammeJob runs on start-up so all we need to do is delete the existing list from the query store.

**Pull request merged**